aten/src/ATen/native/LossNLL.cpp

```diff
@@ -491,7 +491,7 @@ Tensor cross_entropy_loss_prob_target(
 
     switch (reduction) {
       case Reduction::Mean:
-        return -(input * target * weight_).sum() / (input.numel() / input.size(1));
+        return -(input * target * weight_).sum() / ((float) input.numel() / (float) input.size(1));
```

Comment:
Please use C++-style casts (i.e. `static_cast<float>(...)`) over C-style.

Reply:
Changes made and tests added. Apologies for the ridiculous number of commits, I have manage to absolutely butcher a simple change.

Reply:
No worries at all! No change is simple in this codebase tbh :)

Reply:
Thanks :). Anything else you need from me for this one? Do you want me to try and rebase the PR to get rid of these unrelated errors? Or are you happy to review as is?

Reply:
Converted back to draft to fix failing tests

Reply:
All tests now passing. Let me know what you think